cinema/tests/test_movies.py

```diff
@@ -0,0 +1,160 @@
+                    1,
+                ],
+            )
+        )
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_user_access(self):
+        self.client.force_authenticate(self.user)
+
+        res = self.client.get(reverse("cinema:movie-upload-image", args=[1]))
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+        res = self.client.delete(
+            reverse(
+                "cinema:movie-detail",
+                args=[
+                    1,
+                ],
+            )
+        )
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_admin_access(self):
+        self.client.force_authenticate(self.superuser)
+
+        res = self.client.get(reverse("cinema:movie-list"))
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+        res = self.client.get(reverse("cinema:movie-detail", args=[1]))
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+        res = self.client.delete(
+            reverse(
+                "cinema:movie-detail",
+                args=[
+                    1,
+                ],
+            )
+        )
+        self.assertEqual(res.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_movie_list(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(reverse("cinema:movie-list"))
+
+        serializer = MovieListSerializer(Movie.objects.all(), many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_movie_list_with_title_search(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(reverse("cinema:movie-list"), {"title": "Sample1"})
+
+        serializer_movies = MovieListSerializer(
+            Movie.objects.filter(title__icontains="Sample1"), many=True
+        )
+
+        self.assertEqual(res.data, serializer_movies.data)
+
+    def test_movie_list_with_genres_search(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(reverse("cinema:movie-list"), {"genres": ["101", "102"]})
+
+        serializer_movies = MovieListSerializer(
+            Movie.objects.filter(genres__name__icontains=("101", "102")), many=True
+        )
+        self.assertEqual(res.data, serializer_movies.data)
+
+    def test_movie_list_with_actors_search(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(reverse("cinema:movie-list"), {"actors": 1})
+
+        serializer_movie1 = MovieListSerializer(
+            Movie.objects.get(actors__id=1),
+        )
+
+        self.assertIn(serializer_movie1.data, res.data)
+
+    def test_create_movie(self):
+        data = {
+            "id": 700,
+            "title": "teeeest",
+            "description": "teest12345",
+            "duration": 125,
+            "genres": 1,
+            "actors": 1,
```

Comment:
The `genres` and `actors` fields should be lists of IDs, not single integers. Update the test data to reflect this.